R/lint.R

```diff
@@ -446,23 +451,27 @@ check_ranges <- function(ranges, max_col) {
     return()
   }
   if (!is.list(ranges)) {
-    stop("`ranges` must be NULL or a list.", call. = FALSE)
+    cli_abort(c(
+      i = "{.arg ranges} must be {.code NULL} or a {.cls list}.",
+      x = "Instead, it was {.cls {class(ranges)}}."
+    ))
   }
 
   for (range in ranges) {
     if (!is_number(range, 2L)) {
-      stop("`ranges` must only contain length 2 integer vectors without NAs.", call. = FALSE)
+      cli_abort("{.arg ranges} must only contain length 2 integer vectors without {.code NA}s.")
     } else if (!is_valid_range(range, max_col)) {
-      stop(sprintf(
-        "All entries in `ranges` must satisfy 0 <= range[1L] <= range[2L] <= nchar(line) + 1 (%d).", max_col
-      ), call. = FALSE)
+      cli_abort(c(
+        x = "Invalid range specified.",
+        i = "All entries in {.arg ranges} must satisfy 0 <= range[1L] <= range[2L] <= nchar(line) + 1 ({.val {max_col}})."
```

Comment:
@MichaelChirico I'm having a hard time understanding this message. We talk about `ranges`, but the condition to be specified refers to `range`, which is confusing. Can we improve it somehow?

Reply:
Per docs, `ranges` is a list of ranges, i.e. `ranges[[ii]]` needs to be a range.

But we've found `ranges[[ii]][1L] < 0 || ranges[[ii]][1L] > ranges[[ii]][2L] || ranges[[ii][2L] > max_col`, hence the error. We can probably `seq_along()` instead to provide info like:

> ranges[[5]] (2, 1) is not a valid range, which must satisfy 0 <= ranges[1L] <= ranges[2L] <= nchar(line) + 1 (5)

Otherwise we could refactor to something like:

```r
if (!is_number(...)) {
  stop(...)
}
tryCatch(check_valid_range(range,max_col),
  negative_column_error = \(c) stop("ranges[[5]] has lower bound %d < 0"),
  range_decreasing_error = \(c) stop("ranges[[5]] has mixed bounds ranges[1]=%d > %d=ranges[2]"),
  overwide_column_error = \(c) stop("ranges[[5]] has upper bound %d > {max_col}")
)
# or, simpler
tryCatch(check_valid_range(range,max_col),
  range_validation_error = \(c) stop("ranges[[5]]: ", conditionMessage(c))
)
```

Reply:
Thanks for the explanation. 

Out of those options, I like this:

```r
tryCatch(check_valid_range(range,max_col),
  range_validation_error = \(c) stop("ranges[[5]]: ", conditionMessage(c))
)
```